includes/class-wc-stripe-order-handler.php

```diff
@@ -40,6 +42,77 @@ public static function get_instance() {
 		return self::$_this;
 	}
 
+	/**
+	 * Disables the ability to edit for orders with uncaptured payment.
+	 *
+	 * @param $editable boolean The current editability of the order.
+	 * @param $order WC_Order The order object.
+	 * @return boolean false if the order has uncaptured payment, true otherwise.
+	 */
+	public function disable_edit_for_uncaptured_orders( $editable, $order ) {
+		try {
+			$intent = $this->get_intent_from_order( $order );
+
+			if ( $intent && 'requires_capture' === $intent->status ) {
+				$editable = false;
+
+				// add hooks to change text about the reason when order items cannot be edited
+				add_action( 'woocommerce_admin_order_totals_after_total', [ $this, 'maybe_attach_gettext_callback' ] );
+				add_action( 'woocommerce_order_item_add_action_buttons', [ $this, 'maybe_unattach_gettext_callback' ] );
+			}
+		} catch ( Exception $e ) {
+			WC_Stripe_Logger::log( 'Error getting intent from order: ' . $e->getMessage() );
+		}
+
+		return $editable;
+	}
+
+	/**
+	 * Only attach the gettext callback when on admin edit order screen
+	 */
+	public static function maybe_attach_gettext_callback() {
+
+		if ( is_admin() && function_exists( 'get_current_screen' ) ) {
+			$screen = get_current_screen();
+
+			if ( true || ( is_object( $screen ) && in_array( $screen->id, [ 'woocommerce_page_wc-orders', 'edit-shop_order' ], true ) ) ) {
+				// Hook to gettext callback to change the tooltip text
+				add_filter( 'gettext', array( __CLASS__, 'change_order_item_editable_text_tooltip' ), 10, 3 );
+			}
+		}
+	}
+
+	/**
+	 * unattach the gettext callback
+	 */
+	public static function maybe_unattach_gettext_callback() {
+
+		wc_get_logger()->debug( 'maybe_unattach_gettext_callback' );
+
+		if ( is_admin() && function_exists( 'get_current_screen' ) ) {
+			$screen = get_current_screen();
+
+			if ( true || ( is_object( $screen ) && in_array( $screen->id, [ 'woocommerce_page_wc-orders', 'edit-shop_order' ], true ) ) ) {
+				// Unhook gettext callback to prevent extra call impact
+				remove_filter( 'gettext', array( __CLASS__, 'change_order_item_editable_text_tooltip' ), 10 );
+			}
+		}
+	}
+
+	/**
+	* When order items are not editable due to the charge being authorized to capture the current amount,
+	* change the tooltip to explain the reason
+	*/
+	public static function change_order_item_editable_text_tooltip( $translated_text, $text, $domain ) {
+		switch ( $text ) {
+			case 'To edit this order change the status back to "Pending payment"':
+				$translated_text = __( 'This order is no longer editable because the charge has been authorized.', 'woocommerce-gateway-stripe' );
```

Comment:
Please suggest if you can think of a better message.

Reply:
I think this text is fine.

Another option like "This order is not editable because there's a charge pending authorization for this order total. Cancel the order to remove the authorization."

🤷 Might be a bit long idk.